app/controllers/oai_controller.rb

```diff
@@ -93,23 +108,62 @@ def identify_params
     oai_params(:verb)
   end
 
-  # rubocop:disable Metrics/AbcSize
-  def list_record_normalized_dump_candidates
-    @list_record_normalized_dump_candidates ||= begin
-      current_full_dump = @stream.current_full_dump
+  # Get a page of OAI-XML records and a token pointing to the next page
+  def next_record_page(token = nil)
+    # parse the token if we were provided one
+    set, page, from_date, until_date = OaiConcern::ResumptionToken.decode(token) if token
+    page = page.to_i
+
+    # filter normalized dumps and get the corresponding OAI-XML pages
+    # NOTE: each page is guaranteed to have < OAIPMHWriter::max_records_per_file,
+    # but some pages will have exactly that number and others won't. The sequence
+    # isn't predictable; the only thing we promise is that all pages are less than
+    # that size.
+    pages = normalized_dumps(set, from_date, until_date).flat_map(&:oai_xml_attachments)
```

Comment:
I'm eyeing this with a little concern for our larger dumps. Is it worth getting ActiveRecord to do some of this pagination for us, or is it fine in practice?

Reply:
Definitely in favor of pushing more stuff onto the database if possible. How do we do that in ActiveRecord? Is that a Kaminari thing?